src/Type/UnionIterableType.php

```diff
@@ -31,7 +35,7 @@ public function __construct(
 			$throwException();
 		}
 		foreach ($types as $type) {
-			if ($type instanceof IterableType || $type instanceof UnionType) {
```

Comment:
That's intentionally as iterable objects are expected to be included in `$types`. It could be rewritten as `!$type instanceof ObjectType && $type->isIterable() === YES` but I don't think it's worth it.